src/it/scala/temple/containers/GolangSpec.scala

```diff
@@ -26,4 +32,23 @@ abstract class GolangSpec extends DockerShell2HttpService(8081) with DockerTestK
       .asString
       .body
   }
+
+  protected def buildAndValidate(templefile: String): Iterable[String] = {
+    val parsed = DSLProcessor.parse(templefile) fromEither { error =>
+      throw new RuntimeException(error)
+    }
+    val analyzedTemplefile = Analyzer.parseAndValidate(parsed)
+    val detail             = GoLanguageDetail("example.com")
+    val project            = ProjectBuilder.build(analyzedTemplefile, detail)
+
+    analyzedTemplefile.allServices.keys.map(StringUtils.kebabCase).map { service =>
+      val filesInService = project.files.filter { case (file, _) => file.folder.startsWith(service) }
+      val validationErrors = validateAll(
+        filesInService,
+        entryFile = File(service, s"$service.go"),
+      )
+
+      validationErrors
+    }
+  }
```

Comment:
Where’s this come from?

Reply:
It was previously part of the single `simple.temple` test, but we are reusing this logic of building from templefile in a second test, so I've extracted out.